python/pycylon/data/table.pyx

```diff
@@ -198,38 +206,110 @@ cdef class Table:
+        str], op_column_names: List[str]) -> List[int]:
+        resolve_col_ids = []
+        for op_col_id, op_column_name in enumerate(op_column_names):
+            for col_id, column_name in enumerate(column_names):
+                if op_column_name == column_name:
+                    resolve_col_ids.append(col_id)
+        return resolve_col_ids
+
+    def _get_join_column_indices(self, table: Table, **kwargs):
+        ## Check if Passed values are based on left and right column names or indices
+        left_cols = kwargs.get('left_on')
+        right_cols = kwargs.get('right_on')
+        column_names = kwargs.get('on')
+
+        table_col_names_list = [self.column_names, table.column_names]
+
+        if left_cols and right_cols and isinstance(left_cols, List) and isinstance(right_cols,
+                                                                                   List):
+            if isinstance(left_cols[0], str) and isinstance(right_cols[0], str):
+                left_cols = self._resolve_column_indices_from_column_names(self.column_names,
+                                                                           left_cols)
+                right_cols = self._resolve_column_indices_from_column_names(table.column_names,
+                                                                            right_cols)
+                self._check_column_names_viable(left_cols, right_cols)
+
+                return left_cols, right_cols
+            elif isinstance(left_cols[0], int) and isinstance(right_cols[0], int):
+                return left_cols, right_cols
+        ## Check if Passed values are based on common column names in two tables
+        elif column_names and isinstance(column_names, List):
+            if isinstance(column_names[0], str):
+                left_cols = self._resolve_column_indices_from_column_names(self.column_names,
+                                                                           column_names)
+                right_cols = self._resolve_column_indices_from_column_names(table.column_names,
+                                                                            column_names)
+                self._check_column_names_viable(left_cols, right_cols)
+                return left_cols, right_cols
+            if isinstance(column_names[0], int):
+                return column_names, column_names
+        else:
+            raise TypeError("kwargs 'on' or 'left_on' and 'right_on' must be provided")
+
+        if not (left_cols and isinstance(left_cols[0], int)) and not (right_cols and isinstance(
+                right_cols[0], int)):
+            raise TypeError("kwargs 'on' or 'left_on' and 'right_on' must be type List and contain "
+                            "int type or str type and cannot be None")
+    def _is_column_indices_viable(self, left_cols, right_cols):
+        return left_cols and right_cols
+
+    def _check_column_names_viable(self, left_cols, right_cols):
```

Comment:
@nirandaperera @supunkamburugamuve 
This is the function call for check validity of non-zero values for resolved column indices